src/Language/Slicer/Repl.hs

```diff
@@ -56,13 +57,17 @@ parseAndEvalLine line = do
            gamma  <- getGamma
            dsgres <- runSlMIO (desugarEval tyCtx gamma evalS expr)
            case dsgres of
-             Right (val, res, ty, st) ->
+             Right (ORet val, res, ty, st) ->
                  do setEvalState st
                      -- See Note [Handling let bindings]
                     when (isLetBinding expr)
                              (val `seq` addBinding (getVar expr) val ty)
                     return (It $ "val it = " ++ show (pPrint res) ++
                                  " : "       ++ show (pPrint ty))
+             Right (OExn val, _res, _ty, st) ->
+                 do setEvalState st
+                    return (It $ "Exception: " ++ show (val))
+             Right (_) -> error "Unknown outcome (should be impossible)"
```

Comment:
One more comment on this already merged patch. In Haskell it is more common style to avoid parentheses when possible. So typically one would write `Right _` and `show val` instead of `Right (_)` and `show (val)`.